Python/bytecodes.c

```diff
@@ -443,7 +443,7 @@ dummy_func(
             PyStackRef_CLOSE(receiver);
         }
 
-        inst(UNARY_NEGATIVE, (value -- res)) {
+        pure inst(UNARY_NEGATIVE, (value -- res)) {
```

Comment:
Actually, I don't think this is pure for arbitrary input values.

```suggestion
        inst(UNARY_NEGATIVE, (value -- res)) {
```

Reply:
Yeah it's pure for inputs of certain types like you implied. Wonder if we could add a `cond_pure` annotation in the future. Left to another PR I guess.